src-server/main.ts

```diff
@@ -1,32 +1,151 @@
-import express from "express";
-import { env, loadEnv } from "./env";
+import cors from 'cors';
+import express, { Request, Response } from 'express';
+import firebase from './firebaseConfig.js';
+import path from 'path';
+
+import { createServer } from 'vite';
+import { dirname } from 'path';
+import { env, loadEnv } from './env';
+import { fileURLToPath, pathToFileURL } from 'url';
+import {
+  getAuth,
+  createUserWithEmailAndPassword,
+  signInWithEmailAndPassword,
+  signOut,
+  sendEmailVerification,
+} from 'firebase/auth';
+import {
+  getFirestore,
+  collection,
+  doc,
+  setDoc,
+  getDocs,
+  Firestore,
+} from 'firebase/firestore';
 
-// import env variables
 loadEnv();
 
-// create app
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
 const app = express();
+const port = process.env.PORT || 3000;
```

Comment:
It's better if we do not use process.env as it is not typed. If you check the env.ts file, you'll see it exports an object you can use that is fully typesafe and guaranteed.

```suggestion
const port = env.PORT || 3000;
```